### PR TITLE
add resource names in debug cache

### PR DESCRIPTION
**Pull request by ramaraochavali**
Since we are moving hash as cache keys we need a better way to represent them in debug. This PR displays typeUrl and resource names in debug mode.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -330,9 +330,13 @@ func (s *DiscoveryServer) cachez(w http.ResponseWriter, req *http.Request) {
 		writeJSON(w, res)
 		return
 	}
-	keys := s.Cache.Keys()
-	sort.Strings(keys)
-	writeJSON(w, keys)
+	snapshot := s.Cache.Snapshot()
+	resources := make(map[string][]string) // Key is typeUrl and value is resource names.
```
Comment (howardjohn) [inline pilot/pkg/xds/debug.go:334]: we may want similar logic for the "sizes" logic. IMO it's important there as well so we can see which type is using up space

File: pilot/pkg/xds/debug.go [hunk 2]
```diff
@@ -330,9 +330,13 @@ func (s *DiscoveryServer) cachez(w http.ResponseWriter, req *http.Request) {
...
+	for _, resource := range snapshot {
+		resourceType := resource.Resource.TypeUrl
+		resources[resourceType] = append(resources[resourceType], resource.Name)
```
Comment (howardjohn) [inline pilot/pkg/xds/debug.go:337]: maybe `name/hash`? Seems the hash could be useful in some scenarios.. although not as useful as unobfuscated

**Review by hzxuzhonghu — COMMENTED** [review]
Good for debugging, but as 8080 is exposed publicly, i think we should prevent calling from non-local

**Comment by howardjohn** [comment]
It's not exposed publicly anymore and we already have far more sensitive
data (configz)

**Comment by hzxuzhonghu** [comment]
Good to know.

**Comment by ramaraochavali** [comment]
@hzxuzhonghu @howardjohn changed PTAL.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**